src/web/lib/braze/checkBrazeDependencies.ts

```diff
@@ -0,0 +1,100 @@
+import { getBrazeUuid } from '@root/src/web/lib/getBrazeUuid';
+import { hasRequiredConsents } from './hasRequiredConsents';
+import { hideSupportMessaging } from './hideSupportMessaging';
+
+type SuccessResult = {
+	isSuccessful: true;
+	data: ResultData;
+};
+
+type FailureResult = {
+	isSuccessful: false;
+	failureField: string;
+	failureData: any;
+	data: ResultData;
+};
+
+type DependenciesResult = SuccessResult | FailureResult;
+
+type ResultData = { [key: string]: string | boolean };
+
+type DependencyResult = {
+	name: string;
+	value: Promise<string | boolean | null | undefined>;
+};
+const buildFailureResponse = (name: string, value: any, data: ResultData) => ({
+	isSuccessful: false,
+	failureField: name,
+	failureData: value,
+	data,
+});
+
+const buildDependencies = (
+	isSignedIn: boolean,
+	idApiUrl: string,
+): DependencyResult[] => {
+	return [
+		{
+			name: 'brazeSwitch',
+			value: Promise.resolve(window.guardian.config.switches.brazeSwitch),
+		},
+		{
+			name: 'apiKey',
+			value: Promise.resolve(window.guardian.config.page.brazeApiKey),
+		},
+		{
+			name: 'brazeUuid',
+			value: isSignedIn ? getBrazeUuid(idApiUrl) : Promise.resolve(null),
```

Comment:
worth wrapping in a Promise?

Reply:
I might be wrong but I think you can just use the `async` key to do this?

Reply:
So I guess wrapping in a promise would look like this, is this what you had in mind?

```js
new Promise((resolve) => {
  if (isSignedIn) {
    getBrazeUuid(idApiUrl).then(resolve);
  } else {
    resolve(null);
  }
})
```

> I might be wrong but I think you can just use the async key to do this?

I'm interested in this! I _thought_ you could only use `async` to mark a function as async (which as you say means the function will always return a promise). Do you mean use an immediately invoked function in combination with `async` here?

Reply:
It was a drive by comment without fully parsing the code in my head but yes, `async` is like shorthand for 'this function always returns a promise' which I though might be handy here but take my idea with a lot of salt!